### ISSUE TITLE
type-annotation-spacing gets confused about inlined type mappings (on members)

### ISSUE THREAD

**Issue by nevir** [nzakas/eslint-plugin-typescript#69]
With `typescript/type-annotation-spacing: error`, when running typescript-eslint-parser@8.0.0, the following fails:

```ts
interface Foo {
  thing: { [key in string]: number };
  //                     ^ Expected a space after the ']'
  //                     ^ Expected a space before the ']'
}
```

It did not error on typescript-eslint-parser@7.0.0 (not sure if this should be logged on that repo, though?)

---

As a workaround:

```ts
type NumberMap = { [key in string]: number };
interface Foo {
  thing: NumberMap;
}
```

**Comment by weirdpattern** [comment]
I submitted two PRs, one to upgrade the parser and other dependencies in package.json and another to fix this issue and another issue I found affecting the rule with both 7.0.0/8.0.0 parsers.

Basically, the first colon in a class/interface/type property is ignore, making the following valid.

```ts
interface Foo {
  thing:{ [key in string]: number };
       // ^ no space here 
}
```
